mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -160,14 +189,60 @@ public void update(CRUDOperationContext ctx,
                 }
                 docIndex++;
             }
+            // if we have any remaining items to update
+            if (numUpdating > 0) {
+                try {
+                    executeAndLogBulkErrors(bwo);
+                    numFailed += (int) docUpdateErrors.stream().map(e -> e.getIndex()).distinct().count();
```

Comment:
The only reason I do this is because it seems possible for > 1 error to be returned per index, which would skew the actual number failed.  Do you think we should just not worry about that case?